server/athenian/api/db.py

```diff
@@ -391,3 +392,26 @@ async def insert_or_ignore(model,
 def extract_registered_models(base: Any) -> Mapping[str, Any]:
     """Return the mapping from declarative model names to their classes."""
     return base.registry._class_registry
+
+
+@asynccontextmanager
+async def enable_sqlite_foreign_keys(db: Database, conn: Connection) -> AsyncIterator[None]:
+    """Context manager to enable the foreign keys constraints in sqlite for the connection.
+
+    Does nothing if the DB is not sqlite.
+    Must be called outside of any transaction.
+    foreign_keys support is restored when the context manager exits.
+
+    """
+    if db.url.dialect == "sqlite":
+        enabled = await conn.fetch_val("PRAGMA foreign_keys")
```

Comment:
Let's leverage our control in morcilla and unconditionally enable foreign keys this in the sqlite driver. It will be much less code, much less complexity, much more reliability.